### PR TITLE
Fix #180

### LINKED ISSUE #180: `baseUri` defined, but getting "no API endpoint specified"
I'm using abao via `npm`. NPM-Version is: abao@0.4.0

Related issue: #104 

My thought on this: Make `--server` required and implement `--use-default-server` flag later on. My reason for putting this behind a flag is, that it makes hammering production-servers with requests harder. And maybe don't use the default server at all and always require the server to be specified.

jm2c and cheers!

### PR DESCRIPTION
**Pull request by galk-in**
In this pull request fixes for #180

### CODE REVIEW

File: lib/abao.coffee [hunk 1]
```diff
@@ -40,7 +40,7 @@ class Abao
       # Parse tests from RAML
       (raml, callback) ->
         if !config.options.server
-          if 'baseUri' in raml
```
Comment (plroebuck) [inline lib/abao.coffee]: Why was this changed? It was coded as such to denote that `baseUri` is an optional field, per [RAML specs](https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md/#base-uri-and-base-uri-parameters)

Comment (galk-in) [inline lib/abao.coffee]: ```if 'baseUri' in raml``` is compiling in unworking code.
Old compiled javascript was
```javascript
            if (indexOf.call(raml, 'baseUri') >= 0) {
```
as you can check this JS don't pass new tests.
New javascript is 
```javascript
            if (raml.baseUri) {
```
and it's okey with new tests.

If you have an idea why ```if 'baseUri' in raml``` has a bug or have a better idea how to fix it, please do the pull.

File: test/fixtures/single-get.raml [hunk 2]
```diff
@@ -1,8 +1,7 @@
 #%RAML 0.8
 
 title: World Music API
-baseUri: http://example.api.com/{version}
-version: v1
+baseUri: http://localhost:3333
```
Comment (plroebuck) [inline test/fixtures/single-get.raml]: Why was this changed? Seems like ``single-get.raml`` was used by more than one test case. The abstract version is better than this one specific to our test environment setup. Maybe add another file ``single-get-localhost.raml`` for your test, but put the abstract one back like it was.

Comment (galk-in) [inline test/fixtures/single-get.raml]: 👍
